homeassistant/components/sensor/syncthru.py

```diff
@@ -0,0 +1,370 @@
+        default=DEFAULT_NAME
+    ): cv.string,
+    vol.Optional(
+        CONF_MONITORED_CONDITIONS,
+        default=DEFAULT_MONITORED_CONDITIONS
+    ): vol.All(cv.ensure_list, [vol.In(DEFAULT_MONITORED_CONDITIONS)])
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the BLNET component"""
+
+    from pysyncthru import SyncThru, test_syncthru
+
+    if discovery_info is not None:
+        host = discovery_info.get(CONF_HOST)
+        name = discovery_info.get(CONF_NAME, DEFAULT_NAME)
+        # _LOGGER.info("Discovered a new Samsung Printer: %s" % discovery_info)
+        # Test if the discovered device actually is a syncthru printer
+        if test_syncthru(host) is False:
+            _LOGGER.error("No SyncThru Printer reached under given resource")
+            return False
+        # monitor all supported devices
+        monitored = DEFAULT_MONITORED_CONDITIONS
+    else:
+        host = config.get(CONF_RESOURCE)
+        name = config.get(CONF_NAME)
+        # Monitor all devices that are given in the config and are supported
+        monitored = config.get(CONF_MONITORED_CONDITIONS)
+
+    # Main device, always added
+    printer = SyncThru(host)
+    printer.update()
+    devices = [SyncThruSensor(hass, printer, name)]
+
+    for key in printer.tonerStatus(filter_supported=True).keys():
+        if 'toner_' + str(key) in monitored:
+            devices.append(SyncThruToner(hass, printer, name, key))
+    for key in printer.drumStatus(filter_supported=True).keys():
+        if 'drum_' + str(key) in monitored:
+            devices.append(SyncThruDrum(hass, printer, name, key))
+    for key in printer.inputTrayStatus(filter_supported=True).keys():
+        if 'tray_' + str(key) in monitored:
+            devices.append(SyncThruInputTray(hass, printer, name, key))
+    for key in printer.outputTrayStatus().keys():
+        if 'output_tray_' + str(key) in monitored:
+            devices.append(SyncThruOutputTray(hass, printer, name, key))
+
+    add_devices(devices, True)
+    return True
+
+
+class SyncThruSensor(Entity):
+    """Implementation of a Samsung Printer sensor platform."""
+
+    def __init__(self, hass, syncthru, name):
+        """Initialize the sensor"""
+        self._hass = hass
+        self.syncthru = syncthru
+        # init the devices entitiy name starting without number/name
+        self._attributes = {}
+        self._state = STATE_UNKNOWN
+        self._name = name
+        self._icon = 'mdi:printer'
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Return the state of the device."""
+        return self._icon
+
+    def update(self):
+        """Get the latest data from SyncThru and update the state."""
+        syncthru = self.syncthru
```

Comment:
I would remove this assignment and use the `self.syncthru` directly considering it is not that many lines of code.